### ISSUE TITLE
[vue-tsc @ 0.28.8+] Property 'utc' does not exist on type 'typeof dayjs'

### ISSUE THREAD

**Issue by charles-allen** [johnsoncodehk/volar#670]
**TL;DR: I think vue-tsc stopped picking up `.d.ts` from other packages** 

I just bumped the `vue-tsc` version in a previously working `pnpm` monorepo.
- On `0.28.7` (& various older versions) I can build.
- On `0.28.8`, `0.28.9`, `0.28.10` I can't.

```
src/foo/bar.ts:45:29 - error TS2339: Property 'utc' does not exist on type 'typeof dayjs'.

45       const newDate = dayjs.utc(date)
                               ~~~
```

I have imported dayjs as an "initialized" version from a different package:
```ts
import { dayjs } from '@blah/utils/dayjs-initialized'
```

Here is the full contents of that file:
```ts
// This is the one place we import & configure dayjs
import * as dayjsBase from 'dayjs' 
import * as minMax from 'dayjs/plugin/minMax'
import * as pluralGetSet from 'dayjs/plugin/pluralGetSet'
import * as quarterOfYear from 'dayjs/plugin/quarterOfYear'
import * as utc from 'dayjs/plugin/utc' // utc() type comes from this plugin

dayjsBase.extend(utc)
dayjsBase.extend(quarterOfYear)
dayjsBase.extend(pluralGetSet)
dayjsBase.extend(minMax)

export const dayjs = dayjsBase
```

VSCode intellisense & "go to declaration" work fine; I have Volar installed (and I believe working):

![Screen Shot 2021-11-05 at 16 09 21](https://user-images.githubusercontent.com/31621095/140486265-428f283e-5311-4fba-8f75-b3bdd3e07b86.png)

**Comment by charles-allen** [comment]
Just occurred to me you might be interested in my package.json & tsconfig.json:

_package.json script:_
```json
"build": "vue-tsc --project tsconfig.build.json --noEmit && vite build",
```

_tsconfig.build.json_
```json
{
  "extends": "../../tsconfig.base.json",

  "compilerOptions": {
    "baseUrl": "src",
    "outDir": "dist",
    "paths": {
      "@blah/utils/*": ["../../utils/src/*"],
      "@blah/common/*": ["../../common/src/*"],
      "~/*": ["./_services/*"],
      "@/*": ["./*"]
    },

    // Overrides
    "composite": false,
    "incremental": false,
    "types": ["vite/client", "@types/jest", "unplugin-icons/types/vue"],
    "target": "ESNext",
    "jsx": "preserve",
    "resolveJsonModule": true,
    "noUnusedLocals": false // doesn't understand usage in a Vue template
  },

  "include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.vue"],
  "references": [
    { "path": "../utils/tsconfig.build.json" },
    { "path": "../common/tsconfig.build.json" }
  ]
}
```

And for completeness, I guess the IDE / extensions are using `tsconfig.json`
```json
{
  "extends": "./tsconfig.build.json",
  "include": [
    "src/**/*.ts",
    "src/**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "*.config.ts",
    "*.d.ts"
  ],
  "exclude": ["node_modules", "dist", "lib", ".*/**"]
}
```

**Comment by johnsoncodehk** [comment]
Can you make a minimal reproduction? So I can check it quick, thanks.

**Comment by charles-allen** [comment]
Yes, but it will have to be after Wednesday.

**Comment by charles-allen** [comment]
@johnsoncodehk - FYI, we have now seen the same error using `tsc` in a non-vue package. So it seems to be an issue related to using `dayjs` in a typescript monorepo, and not `vue-tsc` specific
